frontend/src/app/chatmenu/SearchMessages.tsx

```diff
@@ -1,19 +1,109 @@
-import React from 'react'
+/* global HTMLTextAreaElement */
+import React, { useState } from 'react'
 import './SearchMessages.css'
-import { ChatMenuPageProps } from './ChatMenuPage'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faMagnifyingGlass } from '@fortawesome/free-solid-svg-icons'
+import { ChatMenuPageProps, MessagesData } from './ChatMenuPage'
+import { Message } from '../chat/Message'
+import { ChatData } from '../types'
 
 interface SearchMessagesProps {
-    // TODO: Fill out the props for your component
+    chat?: ChatData
+    messages?: MessagesData[]
 }
 
-// TODO: Implement your component
-// eslint-disable-next-line no-empty-pattern
-const SearchMessagesInner = ({}: SearchMessagesProps) => {
-    return <p>Nida Akkiswala was here!</p>
+// Extracts the messages in the chat to be filtered through
+const MessagesInner = ({ chat, filteredMessages, totalMessages }) => {
+    let results
+
+    if (filteredMessages?.length === 0) {
+        results = `There were ${filteredMessages?.length} messages with that search. Please try again!`
+    } else if (filteredMessages?.length >= 1) {
+        results = `Showing ${filteredMessages?.length} message(s) from ${totalMessages?.length} messages`
+    }
+
+    return (
+        <>
+            <p>{results}</p>
+            <div>
+                {filteredMessages?.map((message: { key: React.Key | null | undefined }) => (
+                    <Message
+                        key={message.key}
+                        data={message}
+                        myUserId={chat?.for}
+                        participants={chat?.participants}
+                    />
+                ))}
+            </div>
+        </>
+    )
+}
+
+const SearchMessagesInner = ({ messages, chat }: SearchMessagesProps) => {
+    const [value, setValue] = useState<string>()
+    const [filteredMessages, setfilteredMessages] = useState<MessagesData[] | undefined>()
+
+    const handleChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+        setValue(event.target.value)
+    }
+
+    // Conducts the Search in the messages
+    const doSearch = () => {
+        setfilteredMessages(
+            messages?.filter((text) => {
+                const searchString: string = value ?? ''
+
+                return (
+                    text.message.toLowerCase().includes(searchString.toLowerCase()) && // Changes the message to lowercase
+                    searchString !== '' && // Checks for whitespaces
+                    searchString.trim()
+                )
+            })
+        )
+        setValue('')
```

Comment:
Thanks for pointing this out @vingkan 
We did discuss earlier that this could be part of the revisions. We could retain the search string and add `cross` icon to help user clear it prompty, similar to how Google search does it.